src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,121 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K,V>[] table;
+    private int size;
+    private int threshold;
+
+    @SuppressWarnings({"unchecked"})
+    public MyHashMap() {
+        table = (Node<K,V>[])new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int)(DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K,V> node = placeNode(key, value);
+        if (node != null) {
+            node.value = value;
+            return;
+        }
+        if (++size > threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K,V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K,V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(int hash, K key, V value, Node<K,V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private Node<K,V> getNode(K key) {
+        Node<K,V> firstNote = table[(table.length - 1) & hash(key)];
+        Node<K,V> nextNote;
+        int hash = hash(key);
+        if (firstNote != null) {
+            if (firstNote.hash == hash
+                    && (Objects.equals(key, firstNote.key))) {
+                return firstNote;
+            }
+            if ((nextNote = firstNote.next) != null) {
+                do {
+                    if (nextNote.hash == hash
+                            && (Objects.equals(key, nextNote.key))) {
+                        return nextNote;
+                    }
+                } while ((nextNote = nextNote.next) != null);
+            }
+        }
+        return null;
```

Comment:
too complex logic, rewrite it. Algorithm: 
1. find bucket by passed key
2. Go through all nodes in this bucket
- if passed key and key of the current node equal, return value of this current node
3. If a node is not found after traversing through a chain of nodes in a bucket return null